File: lld-problems/Pen/PenTypes/PenFactory.java
package PenTypes;

public class PenFactory {
    public static Writer createPen(PenType penType){
        if(penType == PenType.PEN){
            return new Pen();
        }
        if (penType == PenType.PENCIL) {
            return new Pencil();
        }
        if (penType == PenType.MARKER) {
            return new Marker();
        }
        if (penType == PenType.DIGITAL_PEN) {
            return new DigitalPen();
        }
        return null;
    }
}

File: python-low-level-problems/LLD-Problems/PenProblem/Factory/PenFactory.py
# Factory/PenFactory.py
from Factory.PenType import PenType
from Factory.Pen import Pen
from Factory.Writer import Writer

class PenFactory:
    @staticmethod
    def create_pen(pen_type: PenType) -> Writer:
        if pen_type == PenType.PEN:
            return Pen()
        if pen_type == PenType.PENCIL:
            return Pencil()
        if pen_type == PenType.MARKER:
            return Marker()
        if pen_type == PenType.DIGITAL_PEN:
            return DigitalPen()
        return None
